server/datastore/mysql/software.go

```diff
@@ -659,56 +659,45 @@ func (si *softwareIterator) Next() bool {
 	return si.rows.Next()
 }
 
-func (ds *Datastore) ListSoftwareBySourceIter(
+// AllSoftwareIterator Returns an iterator for the 'software' table, filtering out
+// software entries based on the 'query' param.
+func (ds *Datastore) AllSoftwareIterator(
 	ctx context.Context,
-	sources []string,
+	query fleet.SoftwareIterQueryOptions,
```

Comment:
Nit: Should `AllSoftwareIterator` return empty early (or an error?) if `query.ExcludedSources` contains entries also present in `query.IncludedSources` is not empty? (probably panic or return an error because of how we use this method with fixed/known `query` values).

Reply:
Done in https://github.com/fleetdm/fleet/pull/10514/commits/8c2e90168b77aaf402e96e4b18c5e5bc2ce018c8